### PR TITLE
Task/des 2339 passing new data to datacite

### PR DESCRIPTION
**Pull request by van-go**
## Overview: ##
Update Metadata that is being passed to Datacite
(1) Send Project Title as titleType 'Subtitle'

      "titles": [
    {
      "title": "mission title"
    },
    {
      "title": "Vanessa Field Research",
      "titleType": "Subtitle"
    }
    ],

(2) Send Project description as descriptionType 'Abstract'

     "descriptions": [
    {
      "lang": "en-Us",
      "description": "mission description mission description mission description",
      "descriptionType": "Abstract"
    },
    {
      "lang": "en-Us",
      "description": "description asdf",
      "descriptionType": "Abstract"
    }
    ],

(3) Send Award Title and Number from project level

    "fundingReferences": [
       {
         "awardTitle": "test award",
         "awardNumber": "13245"
       }
    ],

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2339](https://jira.tacc.utexas.edu/browse/DES-2339)

## Summary of Changes: ##
For all project types:
1. Pass project info to to_datacite_json.
2. Include title, description, and award info to datacite metadata.
3. Pass facility info using condition that takes NO facility into account.

## Testing Steps: ##
1. Create a new project or use an old project: (check that all 3 options work)
 - Re-publish 
 - Amend
 - Version
2. Go to DataCite Test (https://doi.test.datacite.org/repositories/tdl.tacc) check for updated data. (Password in Stache)

## Notes: ##
This should work with or without a facility. 
There is a following up ticket that requires modal changes, then updated data to Datacite: [DES-2603](https://jira.tacc.utexas.edu/browse/DES-2603)

### CODE REVIEW

File: designsafe/apps/projects/models/agave/experimental.py [hunk 1]
```diff
@@ -62,19 +62,44 @@ class Experiment(RelatedEntity):
     project = fields.RelatedObjectField(ExperimentalProject)
     dois = fields.ListField('Dois')
 
-    def to_datacite_json(self):
+    def to_datacite_json(self, project={}):
         """Serialize object to datacite JSON."""
         attributes = super(Experiment, self).to_datacite_json()
-        attributes["subjects"] = attributes.get("subjects", []) + [
-            {"subject": self.experimental_facility.title(), }
+        if hasattr(self, 'facility') and len(self.experimental_facility) and ('None' not in self.experimental_facility):
```
Comment (jarosenb) [inline designsafe/apps/projects/models/agave/experimental.py:68]: Can experiments have a `facility` attribute? I think this expression might always evaluate to false.

File: designsafe/apps/projects/models/agave/experimental.py [hunk 2]
```diff
@@ -62,19 +62,44 @@ class Experiment(RelatedEntity):
     project = fields.RelatedObjectField(ExperimentalProject)
     dois = fields.ListField('Dois')
 
-    def to_datacite_json(self):
+    def to_datacite_json(self, project={}):
```
Comment (jarosenb) [inline designsafe/apps/projects/models/agave/experimental.py:65]: Passing lists/dicts as default arguments can lead to strange behavior: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments 

This PR doesn't mutate the project that gets passed in, but someone might come into the codebase later and accidentally do something weird.

**Review by jarosenb — CHANGES_REQUESTED** [review]

**Review by fnets — DISMISSED** [review]

File: designsafe/apps/projects/models/agave/experimental.py (hunk 1 above)
Comment (van-go) [reply designsafe/apps/projects/models/agave/experimental.py:68]: good point. It should be 'experimental_facility'

File: designsafe/apps/projects/models/agave/experimental.py (hunk 2 above)
Comment (van-go) [reply designsafe/apps/projects/models/agave/experimental.py:65]: I updated the code: 
```
def to_datacite_json(self, project=None):
   if project is None:
       project={}
```

[the remark this answers is not in the document]
Comment (van-go) [reply designsafe/apps/projects/models/agave/experimental.py:65]: I ran into an issue in publication.py where to_datacite_json was not sending the project level data. The solution was to pass 'project.' But then I had to pass project to all calls of to_datacite_json.

**Review by jarosenb — APPROVED** [review]
Resolved merge conflicts and tweaked things so that `.to_datacite_json()` can work without passing in a project. Tested and it works for all project types with and without passing a top-level project.

**Pull request merged**